src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -555,67 +703,120 @@ export default function Swap({
               </GreyCard>
             ) : isUnhandledQuoteError(quote?.error) ? (
               <GreyCard style={{ textAlign: 'center' }}>
-                <TYPE.main mb="4px">Error loading quote. Try again later.</TYPE.main>
+                <TYPE.main mb="4px">
+                  <Trans>Error loading quote. Try again later.</Trans>
+                </TYPE.main>
               </GreyCard>
             ) : showApproveFlow ? (
-              <RowBetween>
-                <ButtonConfirmed
-                  buttonSize={ButtonSize.BIG}
-                  onClick={approveCallback}
-                  disabled={approval !== ApprovalState.NOT_APPROVED || approvalSubmitted}
-                  width="48%"
-                  altDisabledStyle={approval === ApprovalState.PENDING} // show solid button while waiting
-                  confirmed={approval === ApprovalState.APPROVED}
-                >
-                  {approval === ApprovalState.PENDING ? (
-                    <AutoRow gap="6px" justify="center">
-                      Approving{' '}
-                      <Loader
-                      // stroke="white"
-                      />
+              <AutoRow style={{ flexWrap: 'nowrap', width: '100%' }}>
+                <AutoColumn style={{ width: '100%' }} gap="12px">
+                  <ButtonConfirmed
+                    buttonSize={ButtonSize.BIG}
+                    onClick={handleApprove}
+                    disabled={
+                      approvalState !== ApprovalState.NOT_APPROVED ||
+                      approvalSubmitted ||
+                      signatureState === UseERC20PermitState.SIGNED
+                    }
+                    // TODO: check width with new v3 design
+                    // width="48%" // GP-WIDTH
+                    width="100%"
+                    altDisabledStyle={approvalState === ApprovalState.PENDING} // show solid button while waiting
+                    confirmed={
+                      approvalState === ApprovalState.APPROVED || signatureState === UseERC20PermitState.SIGNED
+                    }
+                  >
+                    <AutoRow justify="space-between" style={{ flexWrap: 'nowrap' }}>
+                      <span style={{ display: 'flex', alignItems: 'center' }}>
+                        <CurrencyLogo
+                          currency={currencies[Field.INPUT]}
+                          size={'20px'}
+                          style={{ marginRight: '8px', flexShrink: 0 }}
+                        />
+                        {/* we need to shorten this string on mobile */}
+                        {approvalState === ApprovalState.APPROVED || signatureState === UseERC20PermitState.SIGNED ? (
+                          <Trans>You can now trade {currencies[Field.INPUT]?.symbol}</Trans>
+                        ) : (
+                          <Trans>Allow the Uniswap Protocol to use your {currencies[Field.INPUT]?.symbol}</Trans>
```

Comment:
TODO: remember to overwrite `Uniswap` here or in the index

Reply:
done in another PR 👍